cfme/physical/physical_storage.py

```diff
@@ -0,0 +1,227 @@
+    management_networks = SummaryTable(title="Management Networks")
+    relationships = SummaryTable(title="Relationships")
+    power_management = SummaryTable(title="Power Management")
+    firmwares = SummaryTable(title="Firmwares")
+    ports = SummaryTable(title="Ports")
+
+
+class PhysicalStorageDetailsView(ComputePhysicalInfrastructureStoragesView):
+    """Main PhysicalStorage details page."""
+    breadcrumb = BreadCrumb()
+    toolbar = View.nested(PhysicalStorageDetailsToolbar)
+    entities = View.nested(PhysicalStorageDetailsEntities)
+
+    @property
+    def is_displayed(self):
+        title = "{name} (Summary)".format(name=self.context["object"].name)
+        return (self.in_compute_physical_infrastructure_storages and
+                self.breadcrumb.active_location == title)
+
+
+class PhysicalStoragesToolbar(View):
+    """Represents hosts toolbar and its controls."""
+    configuration = Dropdown(text="Configuration")
+    view_selector = View.nested(ItemsToolBarViewSelector)
+
+
+class PhysicalStorageSideBar(View):
+    """Represents left side bar. It usually contains navigation, filters, etc."""
+
+    @View.nested
+    class filters(Accordion): # noqa
+        tree = ManageIQTree()
+
+
+class PhysicalStorageEntitiesView(BaseEntitiesView):
+    """Represents the physical storage view"""
+    @property
+    def entity_class(self):
+        return PhysicalStorageEntity
+
+
+class PhysicalStoragesView(ComputePhysicalInfrastructureStoragesView):
+    toolbar = View.nested(PhysicalStoragesToolbar)
+    sidebar = View.nested(PhysicalStorageSideBar)
+    including_entities = View.include(PhysicalStorageEntitiesView, use_parent=True)
+
+    @property
+    def is_displayed(self):
+        return (self.in_compute_physical_infrastructure_storages and
+                self.title.text == "Physical Storages")
+
+
+@attr.s
+class PhysicalStorage(BaseEntity, Updateable, Pretty, PolicyProfileAssignable, Taggable):
+    """Model of an Physical Storage in cfme.
+
+    Args:
+        name: Name of the physical storage.
+
+    Usage:
+        mystorage = PhysicalStorage(name='sample_Storage')
+        mystorage.create()
+
+    """
+    pretty_attrs = ['name']
+
+    name = attr.ib()
+    provider = attr.ib()
+
+    INVENTORY_TO_MATCH = ['Product Name', 'Serial Number', 'Description',
+                          'Health State', 'Drive Bays', 'Enclosure Count']
+
+    def get_value_from_summary(self, value):
+        """Returns the value of a summary table
+
+        Usage:
+            mystorage = PhysicalStorage(name='sample_Storage')
+            mystorage.get_value_from_summary(value)
+
+        Args:
+            value (str): The Storage property you want to retrieve from Details page.
+        """
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of(value)
+
+    def _wait_for_state_change(self, desired_state, target, provider, view, timeout=300, delay=10):
```

Comment:
Required:
as you used `_` at start indicating it's for internal use. I am not found any use of this method. 

I would not make this as part of the model but used in the test. If still, you like to make part of the model then please avoid the use of arguments like `provider` and `view`. Those you can take from the object only like `self.provider`.